### ISSUE TITLE
Too rigid date parsing logic

### ISSUE THREAD

**Issue by asammartin** [ngageoint/geopackage-core-java#38]
Please fill out as much known and relevant information as possible.

#### Version Information:
  * GeoPackage Java Version: 3.0.1
  * GeoPackage Core Java Version: 3.0.1
  * GeoPackage Core Java Source: mvn central repo
  * IDE Name & Version: Intellij 2018.2
  * Maven Version: (mvn -version) N/A
  * Java Version: (java -version) 1.8.0_161
  * Platform & OS: MacOS 10.13.6
  * Other Relevant Libraries:
  *  * Gradle 4.7

#### Expected Results:

  * What did you expect to happen? 
  *  * I expect more flexible date formatting.
  *  * At very least I expect the library to continue processing the row after reporting the date format error.

#### Observed Results:

  * What happened instead?
exception thrown for invalid date format
  * How often does this occur?
every time i attempt to read the file

#### Output:

  * Any logs, errors, or output messages?
Error importing table drinking_water_supply_management_areas
mil.nga.geopackage.GeoPackageException: Failed to parse date string: 2016/11/18
	at mil.nga.geopackage.db.DateConverter.dateValue(DateConverter.java:144)
	at mil.nga.geopackage.user.UserCoreResultUtils.getValue(UserCoreResultUtils.java:73)
	at mil.nga.geopackage.user.UserResultSet.getValue(UserResultSet.java:179)
	at mil.nga.geopackage.user.UserResultSet.getValue(UserResultSet.java:72)
	at mil.nga.geopackage.features.user.FeatureResultSet.getValue(FeatureResultSet.java:50)
	at mil.nga.geopackage.features.user.FeatureResultSet.getValue(FeatureResultSet.java:14)
	at mil.nga.geopackage.user.UserResultSet.getRow(UserResultSet.java:104)
	at com.palantir.gaia.upload.gpkg.GpkgFeatureImporter.importTable(GpkgFeatureImporter.java:68)

#### Steps to Reproduce:

  1. download gpkg file from https://gisdata.mn.gov/dataset/water-drinking-water-supply
  2. note the date column (DWSA_D) in the feature table (drinking_water_supply_management_areas)
  3. run the code below
  4. observe the failure when reaching `FeatureResultSet resultSet = dao.queryForAll();`

#### Relevant Code:

```java
geoPackage.getFeatureTables().stream().forEach(tableName -> {
    FeatureDao dao = geoPackage.getFeatureDao(tableName);
    FeatureResultSet resultSet = dao.queryForAll();
    while (resultSet.moveToNext()) {
        FeatureRow row = resultSet.getRow();
        ...
    }
});
```

#### Test Files:

  * Links to any files needed for testing?
ftp://ftp.gisdata.mn.gov/pub/gdrs/data/pub/us_mn_state_health/water_drinking_water_supply/gpkg_water_drinking_water_supply.zip

#### Additional Information:

  * Any additional configuration, data, or information that might help with the issue?

**Comment by asammartin** [comment]
i did see that the format was not compliant with the spec but have no control of the GeoPackage files/contents. these changes are exactly what i need. thanks. 

will they make the 3.0.3 release?

**Comment by bosborn** [comment]
Great, yes I'll push them out on the next release, most likely 3.0.3.

**Comment by bosborn** [comment]
@asammartin Here are some builds of the jars you can use for now: [geopackage.zip](https://github.com/ngageoint/geopackage-core-java/files/2250822/geopackage.zip)

**Comment by bosborn** [comment]
Fixed in [3.1.0](https://github.com/ngageoint/geopackage-core-java/releases/tag/3.1.0)
